fabric-network/test/impl/event/transactioneventhandler.js

```diff
@@ -6,241 +6,276 @@
 
 describe('TransactionEventHandler', () => {
-	let stubEventHub;
-	let stubStrategy;
-	let stubTransaction;
-	let stubTransactionID;
+	let strategy;
+	let transaction;
+	let network;
+	let contract;
+	let eventService;
+	let eventServiceManager;
+	let baseListener;
+	let revert;
+	let FakeLogger;
+	let sandbox;
 	const transactionId = 'TRANSACTION_ID';
+	const options = {};
 
-	beforeEach(() => {
-		stubTransactionID = sinon.createStubInstance(TransactionID);
-		stubTransactionID.getTransactionID.returns(transactionId);
-
-		stubTransaction = sinon.createStubInstance(Transaction);
-		stubTransaction.getTransactionID.returns(stubTransactionID);
-
-		stubEventHub = sinon.createStubInstance(ChannelEventHub);
-		stubEventHub.getName.returns('eventHub');
-		stubEventHub.getPeerAddr.returns('eventHubAddress');
-		stubEventHub.registerTxEvent.callsFake((txId, onEventFn, onErrorFn) => {
-			stubEventHub._transactionId = txId;
-			stubEventHub._onEventFn = onEventFn;
-			stubEventHub._onErrorFn = onErrorFn;
-		});
+	let handler;
 
-		stubStrategy = {
-			getEventHubs: sinon.stub(),
-			eventReceived: sinon.stub(),
-			errorReceived: sinon.stub()
+	beforeEach(() => {
+		revert = [];
+		sandbox = sinon.createSandbox();
+		FakeLogger = {
+			debug: () => {
+			},
+			error: () => {
+			},
+			warn: () => {
+			}
 		};
-		stubStrategy.getEventHubs.returns([stubEventHub]);
+		sandbox.stub(FakeLogger);
+		revert.push(TransactionEventHandler.__set__('logger', FakeLogger));
+		eventServiceManager = sinon.createStubInstance(EventServiceManager);
+		eventServiceManager.startEventService.resolves();
+		network = sinon.createStubInstance(Network);
+		network.eventServiceManager = eventServiceManager;
+
+		transaction = sinon.createStubInstance(Transaction);
+		transaction.transactionId = transactionId;
+		contract = sinon.createStubInstance(Contract);
+		contract.network = network;
+		transaction.contract = contract;
+		transaction.getNetwork.returns(network);
+
+		eventService = sinon.createStubInstance(EventService);
+		eventService.name = 'eventService';
+		eventService.unregisterEventListener = sinon.stub();
+		baseListener = sinon.stub();
+		baseListener.eventService = eventService;
+		eventService.registerTransactionListener.returns(baseListener);
+
+		strategy = sinon.createStubInstance(EventStrategy);
+		strategy.getEventServices.returns([eventService]);
+
+		handler = new TransactionEventHandler(transaction, strategy, options);
 	});
 
+
 	afterEach(() => {
+		if (revert.length) {
+			revert.forEach(Function.prototype.call, Function.prototype.call);
+		}
+		sandbox.restore();
 		sinon.restore();
 	});
 
 	describe('#constructor', () => {
-		it('has a default timeout of zero if no options supplied', () => {
-			const handler = new TransactionEventHandler(stubTransaction, stubStrategy);
-			expect(handler.options.commitTimeout).to.equal(0);
+		it('has a default timeout', () => {
```

Comment:
The description has become less informative and it's now testing 3 things instead of 1. Typically testing one aspect of behaviour in each unit test is good.

Reply:
fixed